### PR TITLE
Security fixes and test speed-up

### PR DESCRIPTION
**Pull request by hbdgr**
### **User description**
This PR includes fixes for a few HackenProof reports and some general improvements.
Tests were also refactored. With better use of fixtures, total test time is now about 60% faster.

### What was done

* Faster tests and improved `HyperStakingBase` setup and fixtures
* Added cross-chain destination check in Lockbox for better cross-chain security
* Separated Curve / Superform initialization from core, making deployments easier on chains that do not have them
* Block zero-address strategies
* Added zero-value redeem checks and extra edge-case handling in Allocation `_leave`

  * prevents zero-shares redeem
  * prevents grief-style attacks
  * blocks zero-allocation leave
* Improved storage layout, added slot gaps for safer upgrades
* Increase test coverage

* Additionally setup github CI running solhint checks and unit tests

___

### **PR Type**
Enhancement, Bug fix, Tests

___

### **Description**
- Decouple Superform/Curve init from core

- Add origin-domain check in Lockbox

- Add zero-value redeem/exit validations

- Harden storage layout and gaps

___

### Diagram Walkthrough

```mermaid
flowchart LR
  Core["HyperStakingInit (core init)"] -- "remove Superform/Curve init" --> Integrations["Integration facets init via managers"]
  Lockbox["LockboxFacet"] -- "check origin == destination" --> Security["CC security"]
  Allocation["AllocationFacet"] -- "zero-exit guards" --> Safety["Redeem safety"]
  CurveLib["LibCurve"] -- "router required/set" --> CurveFacet["CurveIntegrationFacet"]
  SuperformLib["LibSuperform"] -- "one-time init + require" --> SuperformFacet["SuperformIntegrationFacet"]
  Tests["Tests"] -- "new edge cases + speed-up" --> Coverage["Higher coverage"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>11 files</summary><table>
<tr>
  <td><strong>HyperStakingInit.sol</strong><dd><code>Remove Superform/Curve init from core initializer</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-b3e675c6bba9a5c5dafc2a95b823bca375c2411df6ba4df104b3b1d28fcb0381">+1/-13</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>CurveIntegrationFacet.sol</strong><dd><code>Require router configured; rename router setter</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-64e778bbcdb20c2ecaa6f497c5df9e2c9f22ccc3080f1dfa78eab8dde23afb8f">+15/-7</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SuperformIntegrationFacet.sol</strong><dd><code>Add storage init, require-initialized guards</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-0e75bd0ceeee7322ec1da342446ae43288704cfbbc4e6fe77d8a6c8480a64d01">+21/-8</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ICurveIntegration.sol</strong><dd><code>Rename router setter in interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-664ea8930f5c3610d74e573bd415af88a641a4af18c4cf1548550078af62c8a0">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ILockbox.sol</strong><dd><code>Add BadLumiaDestination; rename reexecute method</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-ca3149730c6629a09484124f670cd8340c26050f35cd7af8b973a7e2e8da29b5">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ISuperformIntegration.sol</strong><dd><code>Add initializeStorage entrypoint and types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-7831d5e9e848de2bd8d577f8a5e032fe761de27dde3d5b194f904cd22924e611">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LibCurve.sol</strong><dd><code>Add router-required check and setter; shared errors</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-a1be5b391aba5250a8fc8d4c584c527e4ce158017900b71ead547a123bdce1d4">+14/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LibHyperStaking.sol</strong><dd><code>Add storage gaps for upgrade safety</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-e91bc40a6a489969aee5a5d8902bcb5879b958e4874de2e0cc632800e0f7518a">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LibSuperform.sol</strong><dd><code>Add config struct, one-time init, require-initialized</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-ba79e0c53b2879f81a30a9b6bb65694c404eecc2aab759b8cebb93f18e2b5231">+40/-17</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>LibInterchainFactory.sol</strong><dd><code>Add storage gap for LastMessage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-c5f2e801ea3e256a8bda67424f14bbe9c4cd9d549ad14b09e48c4bce57c11ccc">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Errors.sol</strong><dd><code>Centralize shared custom errors and new guards</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-7e5c1613783751d7ab2dec285bb55c8eb4eca93372f44fa50c7d02b26532ba28">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td><strong>AllocationFacet.sol</strong><dd><code>Add zero-exit checks and edge-case handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-693e2d3ffbcf7103e2077db3c3cbcb6d2d62e7a0c7bd39dd6dfb20256d439c27">+11/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>HyperFactoryFacet.sol</strong><dd><code>Validate nonzero strategy on addStrategy</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-d54ef18bef7a611612eba919ce07aa15f630f5d36ec676e4a961401d89974468">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>LockboxFacet.sol</strong><dd><code>Enforce origin-domain; rename reexecute function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-38de5ed27d598def25a61ecb13c74521696c3cf2b4ab3e680dcf9ecd4fb4c750">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>RealAssetsFacet.sol</strong><dd><code>Validate nonzero addresses and shares on redeem</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-0d4819bb8120bb6f1763c2293d39a84158297ff0f32040c7dc13a85659e0d915">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>HyperlaneHandlerFacet.sol</strong><dd><code>Import shared BadOriginDestination error</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-53bc7e4d919b8231f3e1bb31a5040a36c953da82029a3b787b6ad1d9150bfd27">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>IHyperlaneHandler.sol</strong><dd><code>Remove local BadOriginDestination declaration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-1617476bed7fdcf54c4951cbbf6287ed695b861968e3bc31017f30882c85a9fc">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td><strong>TestSwapIntegration.sol</strong><dd><code>Use new SuperformConfig and Curve setter</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-3b92a413a3f5419a14a9fa58661b8978384166a155746b6876e54efbd323077d">+9/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>SwapStrategy.test.ts</strong><dd><code>Refactor helpers; import stableUnits</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-3c27b00563aa6c77551d97f957c3afcf731af928a458cc31f01ce03f2d7db3a3">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>setup.ts</strong><dd><code>Remove Superform/Curve from base deploy</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-d28a461c46dd3d6440988800fbf174825ddbec8b60488095df612d33f1cad4df">+2/-28</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>shared.ts</strong><dd><code>Add helpers for Curve and shared errors iface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-f0abd6b40d03656b70fef247b9b3e96f5ae705ebfa15fbaf5d472dd366c1b993">+31/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Aera.test.ts</strong><dd><code>Use fixtures; add edge-case redeem test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-c911fbb2d38aba59611e16bfab174cb2e4f440730c65726302b67d7f5827f379">+63/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>CurveStrategy.test.ts</strong><dd><code>Initialize integrations in tests; new router guards</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-e9749831b546fbbfe5fbcca5ab2eb50604e27f570701698fadcc8c0ff2a1ef2d">+73/-5</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Diamond.test.ts</strong><dd><code>Consolidate typechain imports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-f187a91d2d61ef5b18cd7eb625ae953209f94b810e4cad61481a69b982f603a6">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>DiamondCacheBug.test.ts</strong><dd><code>Consolidate typechain imports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-326ec028a7fd4575f2a042a122311bb2bada0f7cd33537d3bebe2d27cbcb6e07">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Lockbox.test.ts</strong><dd><code>Rename reexecute API; add origin check tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-6bf32ccf722dd4a074769d2444603d178707576c42fc36af1eb36d5566deefba">+98/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Staking.test.ts</strong><dd><code>Use fixtures for deploy</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-77fa49ac9fd5c2c1d692693b94f8b11fe7b00e73a1775514873d50050298fa2b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Strategy.test.ts</strong><dd><code>Use fixtures; add ZeroAddress strategy test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-f3691333309d201454d051bfe35499ce0f8e17d0598dad6d64bf85c30d08da47">+17/-4</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>Superform.test.ts</strong><dd><code>Add init-required tests; config-based setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-a72c16af620bdf4e906c2375484069dd047740fb711a44bd9ee7e34f63a50261">+83/-6</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>VaultShares.test.ts</strong><dd><code>Add zero-value redeem validations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-0f35510ce144af05ecf03bb38394b577fb303645e7da6588863ee9536dbf29e5">+48/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>HyperStaking.ts</strong><dd><code>Drop Superform/Curve params from init call</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-4cde7e6f48205dc7321710d769e678a7bab55bb5fca283f34b3b241584a57bcd">+0/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>parameters.holesky.json</strong><dd><code>Remove Superform params from HyperStaking config</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/LumiaChain/HyperStaking-contracts/pull/6/files#diff-7cd54174462fb789231521f74c9a3f37d31f959d174686722dbb035776fb97a7">+1/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

> [!NOTE]
> Decouples Curve/Superform initialization into integration facets with guards, adds strict origin-domain checks, hardens redeem/exit validations, centralizes errors, updates storage, refactors tests, and adds CI.
> 
> - **Core/Integrations**:
>   - Remove Curve/Superform setup from `HyperStakingInit`; initialize via `SuperformIntegrationFacet.initializeStorage` and `CurveIntegrationFacet.setCurveRouter` with `requireInitialized`/`requireRouter` guards.
>   - Rename APIs: `updateCurveRouter` -> `setCurveRouter`, `reexecuteStakeRedeem` -> `reexecuteFailedRedeem`.
> - **Cross-chain security**:
>   - Enforce origin-domain validation in `LockboxFacet.handle` and `HyperlaneHandlerFacet.handle` (`BadOriginDestination`).
> - **Validation/Guards**:
>   - `AllocationFacet._leave`: block zero-allocation and zero-asset exits (`ZeroAllocationExit`, `ZeroStakeExit`).
>   - `RealAssetsFacet.redeem`: reject zero addresses/amounts.
>   - `HyperFactoryFacet.addStrategy`: reject zero-address strategies.
> - **Libraries/Storage/Errors**:
>   - `LibSuperform`: add one-time init, `requireInitialized`.
>   - `LibCurve`: add `requireRouter`, `setRouter`.
>   - `LibHyperStaking`, `LibInterchainFactory`: add storage gaps.
>   - Introduce shared `contracts/shared/Errors.sol`; update imports.
> - **Config/CI**:
>   - Add GitHub Actions CI for lint, solhint, unit tests (`.github/workflows/test_unit.yml`).
>   - `solhint`: disable `import-path-check`.
>   - Update ignition params/modules to drop integration init args.
> - **Tests**:
>   - Heavy fixture refactor; add edge-case tests for cross-chain checks, zero exits, router/initialization guards; helper utilities for Curve/Superform; faster overall execution.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d8f16cd003febcf90f926030e2b75386bfced2bd. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: contracts/hyperstaking/facets/AllocationFacet.sol [hunk 1]
```diff
@@ -202,8 +203,16 @@ contract AllocationFacet is IAllocation, HyperStakingAcl, ReentrancyGuardUpgrade
...
+        allocation = need;
+        if (need > capUnits) {
+            // edge-case: check ZeroStakeExit
+            require(IStrategy(strategy).previewExit(allocation) > 0, ZeroStakeExit());
+            // min(need, capUnits) also fix +1 ceil from previewAllocation
+            allocation = capUnits;
+        }
+
+        // edge-case: prevent zero shares exit
+        require(allocation > 0, ZeroAllocationExit());
```
[the remark this answers is not in the document]
Comment (hbdgr) [reply contracts/hyperstaking/facets/AllocationFacet.sol]: done

File: contracts/hyperstaking/interfaces/ILockbox.sol [hunk 2]
```diff
@@ -47,6 +47,7 @@ interface ILockbox {
 
     error NotFromMailbox(address from);
     error NotFromLumiaFactory(address sender);
+    error BadLumiaDestination(uint32 lumiaDestination);
```
[the remark this answers is not in the document]
Comment (hbdgr) [reply contracts/hyperstaking/interfaces/ILockbox.sol]: removed

File: contracts/hyperstaking/facets/AllocationFacet.sol [hunk 3]
```diff
@@ -202,9 +203,17 @@ contract AllocationFacet is IAllocation, HyperStakingAcl, ReentrancyGuardUpgrade
...
 
+        // edge-case: prevent zero shares exit
+        require(allocation > 0, ZeroAllocationExit());
+
+        // edge-case: check if exit will result in zero assets
+        require(IStrategy(strategy).previewExit(allocation) > 0, ZeroStakeExit());
+
+        // edge-case: prevent zero shares exit
+        require(allocation > 0, ZeroAllocationExit());
+
```
[the remark this answers is not in the document]
Comment (hbdgr) [reply contracts/hyperstaking/facets/AllocationFacet.sol]: fixed

**Comment by dnzdlklc** [comment]
bugbot run
